src/reducers/sources.js

```diff
@@ -74,6 +74,10 @@ function update(
     }
 
     case "SELECT_SOURCE":
+      if (action.status != "start") {
+        return state;
+      }
+
```

Comment:
select source did get a bit more complicated, but i think it is better now. i.e. we want to know when we're done selecting the source...